src/Select.tsx

```diff
@@ -16,15 +17,41 @@ export * from './generated/Select.js';
 
 export type SelectReactRendererProps = ReactSimpleRendererProps<SelectElement>;
 
+type SelectRenderer = ComponentType<SelectReactRendererProps>;
+
 export type SelectProps = Partial<Omit<_SelectProps, 'children' | 'renderer'>> &
   Readonly<{
-    children?: ReactNode | ComponentType<SelectReactRendererProps>;
-    renderer?: ComponentType<SelectReactRendererProps> | null;
+    children?: ReactNode | SelectRenderer | Array<ReactNode | SelectRenderer>;
+    renderer?: SelectRenderer | null;
   }>;
 
 function Select(props: SelectProps, ref: ForwardedRef<SelectElement>): ReactElement | null {
+  const children = Children.toArray(props.children as ReactNode[]);
+
+  // Components with slot attribute should stay in light DOM.
+  const slotted = children.filter((child: any) => {
+    return 'props' in child && child.props.slot;
+  });
+
+  // Component without slot attribute should go to the overlay.
+  let overlayChildren: ReactNode[] | undefined;
+
+  children.forEach((child) => {
+    if (!slotted.includes(child)) {
+      if (!overlayChildren) {
+        overlayChildren = [];
+      }
+      overlayChildren.push(child);
+    }
+  });
```

Comment:
nit: Could simplify this to:
```ts
const overlayChildren = children.filter((child) => !slotted.includes(child));
```

and then pass `renderFn || overlayChildren.length ? overlayChildren : undefined` to the renderer hook

Reply:
Done.